src/core/grid_based_algorithms/lb_interface.cpp

```diff
@@ -208,15 +208,20 @@ void lb_lbfluid_propagate() {
 void lb_lbfluid_on_integration_start() {
   if (lattice_switch == ActiveLB::GPU) {
 #ifdef LB_GPU
-    lb_GPU_sanity_checks();
-    if (this_node == 0 && lb_reinit_particles_gpu()) {
-      lb_realloc_particles_gpu();
-      lb_reinit_particles_gpu.validate();
+    if (this_node == 0) {
+      lb_GPU_sanity_checks();
+      lb_boundary_mach_check();
+
+      if (lb_reinit_particles_gpu()) {
+        lb_realloc_particles_gpu();
+        lb_reinit_particles_gpu.validate();
+      }
     }
 #endif
   } else if (lattice_switch == ActiveLB::CPU) {
 #ifdef LB
     lb_sanity_checks();
+    lb_boundary_mach_check();
```

Comment:
I'd prefer to have a central interface function like ```lb_lbfluid_sanity_checks``` that decides which functions to call (either CPU or GPU related)